File: Algorithms/532. K-diff Pairs in an Array/swift/solution.swift
class Solution {
    func findPairs(_ nums: [Int], _ k: Int) -> Int {
        var counter = [Int: Int]()
        for num in nums {
            counter[num, default: 0] += 1
        }

        var result = 0
        for (num, count) in counter {
            if (k == 0 && count > 1) || (k > 0 && counter[num - k] != nil) {
                result += 1
            }

        }

        return result
    }
}

File: Algorithms/532. K-diff Pairs in an Array/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return Integer
     */
    function findPairs($nums, $k)
    {
        $counter = array_count_values($nums);

        $result = 0;
        foreach ($counter as $num => $count) {
            if (($k > 0 && isset($counter[$k + $num])) || ($k == 0 && $count > 1)) {
                $result++;
            }
        }
        return $result;
    }
}
